pkg/processor/build/builder.go

```diff
@@ -516,6 +516,11 @@ func (b *Builder) validateAndEnrichConfiguration() error {
 		b.options.FunctionConfig.Spec.Runtime = "python:3.9"
 	}
 
+	// enrich project name
+	if _, err := b.options.FunctionConfig.GetProjectName(); err != nil {
+		b.options.FunctionConfig.Meta.Labels[common.NuclioResourceLabelKeyProjectName] = platform.DefaultProjectName
```

Comment:
Doesn't this already happen in the enrichment phase?
In what flow will we get here?

Reply:
@TomerShor in `nuctl`

Reply:
I see.
IMO the enrichment should happen before and not be the Builder's responsibility (SoC). It can happen in `nuctl/build.go`

Reply:
@TomerShor I see your point for sure, but why a project is different from, for example, runtime, which we also enrich in `Builder`?

Reply:
Fair point, and this is the reason we enrich again:
https://github.com/nuclio/nuclio/blob/c596b3c994ba66f3c98efadbd550f40b5dde892e/pkg/processor/build/builder.go#L243-L245

Reply:
@TomerShor yeah, there are indeed a lot of things to align, here is one of the places where we enrich things as well - https://github.com/nuclio/nuclio/blob/364c83d37a0632748e23b332e0b37b0ce44f118e/pkg/nuctl/command/deploy.go